api/hardwaremanagement/v1alpha1/node_pools.go

```diff
@@ -42,6 +42,9 @@ type NodePoolSpec struct {
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Location Spec",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
 	LocationSpec `json:",inline"`
 
+	// HwMgrId is the identifier for the hardware manager plugin adaptor.
+	HwMgrId string `json:"hwMgrId" yaml:"hwMgrId"`
```

Comment:
This breaks the current plugin code, as it is adding a new *required* field. I think add `omitempty` here may avoid that. We can likely drop the `omitempty` later once the plugin is updated to know about the new field.

Reply:
Verified that adding `omitempty` allows the current plugin code to work, which is referencing the previous version of the CRD, which it causes a failure to update the CR without it, as it would be missing this new field.
```suggestion
	HwMgrId string `json:"hwMgrId,omitempty"`
```

Reply:
Right, need to support the current plugin code.

Reply:
Updated.